common/wivrn_packets.h

```diff
@@ -218,7 +218,14 @@ struct feedback
 	uint8_t times_displayed;
 };
 
-using packets = std::variant<headset_info_packet, feedback, audio_data, handshake, tracking, hand_tracking, inputs, timesync_response>;
+struct battery
+{
+	float charge;
+	bool present;
+	bool charging;
+};
```

Comment:
Is there an API in Android to query controller battery as well? If we want to support it in the future, we will need a device id associated with the data and maybe bundle multiple statuses in one packet.

Reply:
the only way to do it is via the old oculusvr api afaik.